pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/DoubleMutableDictionary.java

```diff
@@ -31,42 +40,32 @@ public void index(Object rawValue) {
       return;
     }
 
-    if (rawValue instanceof String) {
-      Double entry = Double.parseDouble((String) rawValue);
-      addToDictionaryBiMap(entry);
-      updateMinMax(entry);
-      return;
-    }
-
-    if (rawValue instanceof Double) {
-      addToDictionaryBiMap(rawValue);
-      updateMinMax((Double) rawValue);
-      return;
-    }
-
     if (rawValue instanceof Object[]) {
```

Comment:
should we not reorder the if conditions so that happy path is not expensive. For example, for Double dictionary, it would check if rawValue is Double first and then String. We can also know if its single value or multi value from the schema and avoid checking for array type or atleast push array type to the end.